### PR TITLE
FieldDisplayName: Make field display names unique if field name exists more than once

### LINKED ISSUE #27781: Transform outer join - error when the metric name of the two tables is the same
I have Grafana 7.1.5 and the data is in Elasticsearch 7.X 
When make two queries to apply transform outer join from one field, but the join repeat the values of one table twice because the metric name is the same.
The first query is to count when have one condition and the second for another condition like "error" and "correct"

image

if not make the outer join the values of counter are different  between the two queries:

image

image

If tried to make transform organize fields, only appear one metric, because the metric fields have the same name:

image

### PR DESCRIPTION
**Pull request by torkelo**
This was a bit trickier than I thought. And not sure if this logic is correct or what side effects it has (existing transforms seem to work without modification). 

I had to modify seriesToColumns as MutableDataFrame.addField does not add the instance you pass to the function but creates a new field instance. 

Given two data frames with fields:
type, value 

Join now gives us: 

Screenshot from 2020-10-01 14-08-05

Fixes #27781

Alternative to #27931

### CODE REVIEW

File: packages/grafana-data/src/transformations/transformers/seriesToColumns.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export const seriesToColumnsTransformer: DataTransformerInfo<SeriesToColumnsOpti
     });
 
     for (const item of allFields) {
-      resultFrame.addField(item.newField);
+      item.newField = resultFrame.addField(item.newField);
```
Comment (hugohaggmark) [inline packages/grafana-data/src/transformations/transformers/seriesToColumns.ts:71]: Maybe we should add the same test case from my PR?

**Review by hugohaggmark — APPROVED** [review]
LGTM! I think we should copy the test case from my PR https://github.com/grafana/grafana/blob/176cb648ac2c0cfc584280607301102e7ef9e1c6/packages/grafana-data/src/transformations/transformers/seriesToColumns.test.ts#L351

File: packages/grafana-data/src/transformations/transformers/seriesToColumns.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-data/src/transformations/transformers/seriesToColumns.ts:71]: yea, I need to add a test case for seriesToColumn, as without the above line things break quite badly

Comment (torkelo) [reply packages/grafana-data/src/transformations/transformers/seriesToColumns.ts:71]: added test

**Comment by torkelo** [comment]
Not sure most have, but some definitely.

And for a tranformer like organize fields it has definitely. And that is the main way to rename fields , and it currently cannot distinguish between two fields with same display name

**Comment by torkelo** [comment]
and without unique names, the whole field overrides system sort of does not work

**Review by ryantxu — APPROVED** [review]
:shrug: likely better to have this constraint

**Review by mckn — APPROVED** [review]
I think this looks good! Great work @torkelo!

If I don't recall incorrectly the transforms I have been involved in will probably pick the first time field as the default one (if they have the same name). This should not have an impact on the merge transform since it uses the `field.name`.

**Pull request merged**